src/util/records.py

```diff
@@ -410,8 +410,33 @@ def remove_records(cr, model, ids):
     table = table_of_model(cr, model)
     cr.execute('DELETE FROM "{}" WHERE id IN %s'.format(table), [ids])
     for ir in indirect_references(cr, bound_only=True):
-        query = 'DELETE FROM "{}" WHERE {} AND "{}" IN %s'.format(ir.table, ir.model_filter(), ir.res_id)
-        cr.execute(query, [model, ids])
+        if ir.company_dependent_comodel:
+            if ir.company_dependent_comodel == model:
+                cr.execute(
```

Comment:
We could even explode this UPDATE query.
//cc @aj-fuentes

Reply:
Indeed it must be exploded.